crates/cranelift/src/func_environ.rs

```diff
@@ -2369,25 +2361,23 @@ impl<'module_environment> crate::translate::FuncEnvironment
             }
         };
 
-        let page_size_log2 = memory.page_size_log2;
+        let heap_bound = func.create_global_value(ir::GlobalValueData::Load {
+            base: ptr,
+            offset: Offset32::new(current_length_offset),
+            global_type: pointer_type,
+            flags: MemFlags::trusted(),
+        });
```

Comment:
It's this extra global that is causing the changes in the disassembly tests, right? E.g., `gv5`?

Reply:
Right yeah this is the global being added. I [wrote up a bit above about this](https://github.com/bytecodealliance/wasmtime/pull/9543#issue-2630163393) but it's essentially easier given the current code shape to assume that the global value is always there rather than calculate if it'll be needed in a function body and create it and/or lazily creating it on-demand. My hope is that the "overhead" here is just a few bytes in a vector per-function which in theory doesn't have too much impact, but I'll admit I haven't measured.